worlds/tunic/grass.py

```diff
@@ -0,0 +1,6442 @@
+    "Guardhouse 2 - Guard House 2 Upper Grass (3) (155.5, 0.0, -13.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (32) (156.5, 0.0, -14.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Bush (9) (153.0, 0.0, -11.0)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (4) (152.5, 0.0, -12.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (10) (151.5, 0.0, -12.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (13) (151.5, 0.0, -9.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Bush (8) (151.0, 0.0, -11.0)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (8) (150.5, 0.0, -12.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (15) (150.5, 0.0, -9.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (35) (156.5, 0.0, -15.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (29) (155.5, 0.0, -19.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (30) (155.5, 0.0, -18.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (31) (154.5, 0.0, -19.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (28) (154.5, 0.0, -18.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (33) (157.5, 0.0, -15.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (34) (157.5, 0.0, -14.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Bush (3) (157.0, 0.0, -13.0)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (37) (157.5, 0.0, -11.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (38) (157.5, 0.0, -10.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (39) (156.5, 0.0, -11.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Grass (36) (156.5, 0.0, -10.5)": TunicLocationData("Guard House 2 Upper", "Guard House 2 Upper"),
+    "Guardhouse 2 - Guard House 2 Upper Bush (2) (155.0, 0.0, -11.0)": TunicL
```

Comment:
classic optimization review:tm:
it's expensive to check options in access_rules, and since you're setting can_break_grass in this loop you could just check for start_with_sword here instead of in can_break_grass so fill never has to check the option
I wouldn't be worried about it if it wasn't so easy to apply it

Reply:
those,, weren't the lines i highlighted,,,
the loop where you add start_with_sword is the part i'm commenting on LOL

Reply:
Which is more expensive: checking `state.has_any_count({"Sword": 1, "Sword Upgrade": 2}, player)` or checking the start with sword option?

Reply:
(tried really hard to) show what I'm suggesting in the below comment